### PR TITLE
Update GitHub Action for BCR Reviewer

### PR DESCRIPTION
**Pull request by meteorcloudy**
- Merged the module maintainer notifier and the PR approval into one
  action, which can be distinguished by the "action-type" input.
- Implement a new functionality to dismiss stale approvals from all reviewers. 
  The default `Dismiss stale pull request approvals when new commits are pushed` option from GitHub doesn't dismiss approvals from users who is not a repo maintainer. This helps address the concern that a local commit was made before the approval but pushed after it. See https://github.com/bazelbuild/bazel-central-registry/pull/1549#discussion_r1506081777

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Tested in https://github.com/meteorcloudy/bazel-central-registry/pull/20 with configs located at https://github.com/meteorcloudy/bazel-central-registry/tree/main/.github/workflows

**Comment by Wyverald** [comment]
The 'dismiss review' config LGTM! (for posterity: https://github.com/meteorcloudy/bazel-central-registry/blob/main/.github/workflows/dismiss-approvals.yml)

**Review by Wyverald — APPROVED** [review]

File: actions/bcr-pr-reviewer/index.js
Comment (meteorcloudy) [reply actions/bcr-pr-reviewer/index.js:96]: Yeah, ideally it should fallback to notify BCR maintainers, but I don't think it's that important. BCR maintainers usually go through the open PRs anyway.

**Pull request merged**
